src/platform/packages/shared/response-ops/rule_form/src/rule_details/rule_dashboards.tsx

```diff
@@ -121,14 +131,14 @@ export const RuleDashboards = ({ contentManagement }: Props) => {
   const loadDashboards = useCallback(async () => {
     if (contentManagement) {
       const dashboards = await dashboardServiceProvider(contentManagement)
-        .fetchDashboards()
+        .fetchDashboards({ limit: 1000, text: `${searchValue}*` })
         .catch(() => {});
       const dashboardOptions = (dashboards ?? []).map((dashboard: DashboardItem) =>
         getDashboardItem(dashboard)
       );
       setDashboardList(dashboardOptions);
     }
-  }, [contentManagement]);
+  }, [contentManagement, searchValue]);
 
   useMemo(() => {
     loadDashboards();
```

Comment:
This is probably out of scope for this PR, but I also wonder if it would be better to only load the dashboards if the combo box is selected? Right now we are pre-populating the combo box every time the create rule or edit rule is opened.

Reply:
Draft PR [here](https://github.com/elastic/kibana/pull/223308)